evennia/commands/default/building.py

```diff
@@ -2308,10 +2310,104 @@ class CmdExamine(ObjManipCommand):
+            elif isinstance(value, str):
+                if attrname:
+                    return "\n%s\"%s\": \"%s\"" % (inner_indent, attrname, value)
+                else:
+                    return "\n%s\"%s\"" % (inner_indent, value)
+            elif isinstance(get_data(value), (list, tuple, set)):
+                if attrname:
+                    output += "\n%s\"%s\"%s" % (inner_indent, attrname, get_class_name(value))
+                else:
+                    output += "\n%s%s" % (inner_indent, get_class_name(value))
+
+                # handling for sets
+                if value is set:
+                    output += "{"
+                else:
+                    output += "["
+
+                index = 0
+                for inner_value in value:
+                    output += recursive_print(None, inner_value, depth + 1, indent, references)
+                    index += 1
+                    if index < len(value):
+                        output += ","
+                output += "\n%s" % inner_indent
+
+                # more handling for sets
+                if value is set:
+                    output += "}"
+                else:
+                    output += "]"
+            elif hasattr(value, 'db_key') and hasattr(value, 'dbref'):
+                # This is an ObjectDB thing.
+                if attrname:
+                    return "\n%s\"%s\"%s<%s (%s)>" % (
+                        inner_indent,
+                        attrname,
+                        get_class_name(value),
+                        value.db_key,
+                        value.dbref
+                    )
+                else:
+                    return "\n%s%s <%s (%s)>" % (inner_indent, get_class_name(value), value.db_key, value.dbref)
+            elif value is None:
+                if attrname:
+                    return "\n%s\"%s\": None" % (inner_indent, attrname)
+                else:
+                    return "\n%sNone" % inner_indent
+            else:
+                # Non-primitive type.
+                references[id(value)] = value
+                output += "\n%s\"%s\"%s{" % (inner_indent, attrname, get_class_name(value))
+                data = get_data(value)
+                index = 0
+
+                try:
+                    for key in data.keys():
+                        index += 1
+                        if id(data[key]) in references:
+                            output += "\n%s%s: (ref: %s)" % (inner_indent, key, id(data[key]))
+                            if index < len(data.keys()):
+                                output += ","
+                            continue
+                        output += recursive_print(key, data[key], depth + 1, indent, references)
+                        if index < len(data.keys()):
+                            output += ","
+                except AttributeError:
+                    return "\n%s\"%s\"%s #-1 Cannot Render" % (inner_indent, attrname, get_class_name(value))
```

Comment:
Rather than failing here, it seems more useful to fall back to the normal examine command's mechanism, which works for all of these (although not as prettily)